csvimport_connector.py

```diff
@@ -0,0 +1,330 @@
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(phantom.APP_ERROR, "Unable to find container: {}".format(message))
+
+        # run_automation = param.get('run_automation', True)
+        if vault_id:
+            ret_val, file_info = self._get_file_info_from_vault(action_result, vault_id)
+            if phantom.is_fail(ret_val):
+                return ret_val
+            self.debug_print("File Info", file_info)
+            self.save_progress("****File Info : {0}".format(file_info))
+            csv_vault_path = file_info['path']
+            with open(csv_vault_path, "r") as csvf:
+                reader = csv.reader(csvf)
+                for row in reader:
+                    cef = {}
+                    for i in range(num_columns):
+                        cef[cef_names[i]] = row[i]
+                    phantomrules.add_artifact(
+                        container=container_id, raw_data={}, cef_data=cef, label=artifact_label,
+                        name=artifact_name, severity='high',
+                        identifier=None,
+                        artifact_type=artifact_label)
+                    # self.save_progress("****JSON : {0}".format(artifact_json))
+        action_result.add_data({'vault_id': vault_id})
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _validate_integer(self, action_result, parameter, key, allow_zero=False):
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return action_result.set_status(phantom.APP_ERROR, CSVIMPORT_INVALID_INT.format(param=key)), None
+
+                parameter = int(parameter)
+            except Exception:
+                return action_result.set_status(phantom.APP_ERROR, CSVIMPORT_INVALID_INT.format(param=key)), None
+
+            if parameter < 0:
+                return action_result.set_status(phantom.APP_ERROR, CSVIMPORT_ERR_NEGATIVE_INT_PARAM.format(param=key)), None
+            if not allow_zero and parameter == 0:
+                return action_result.set_status(phantom.APP_ERROR, CSVIMPORT_ERR_INVALID_PARAM.format(param=key)), None
+
+        return phantom.APP_SUCCESS, parameter
+
+    def handle_action(self, param):
+
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'ingest_csv':
+            ret_val = self._handle_ingest_csv(param)
+        if action_id == 'create_csv':
+            ret_val = self._handle_create_csv(param)
+
+        return ret_val
+
+    def initialize(self):
+
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+
+        self._base_url = 'https://127.0.0.1'
```

Comment:
Avoid usage of `127.0.0.1`, instead use `CsvImportConnector._get_phantom_base_url()`.